source/js/weekly_navigation.js

```diff
@@ -63,30 +180,117 @@ function populate(key) {
+        }
+        console.log(tempKey + ", " + test_events[tempKey].name);
+        const allEve = test_events[tempKey].events;
+        const day = test_events[tempKey].name;
+        console.log(allEve);
+
+        for(let j = 0; j < allEve.length; j += 1 ){
+            const event_label = document.createElement('label');
+
+            const start_time = allEve[j].from;
+            const end_time = allEve[j].to;
+
+            const start_time_hr = start_time.slice(0, start_time.indexOf(':'));
+            const start_time_minutes = start_time.substr(start_time.indexOf(':')+1, 3);
+            
+            const end_time_hr = end_time.slice(0, end_time.indexOf(':'));
+            const end_time_minutes = end_time.substr(end_time.indexOf(':')+1, 3);
+            
+            const start_time_ampm = start_time.slice(-2);
+            const end_time_ampm = end_time.slice(-2);
+
+            console.log(`start time hr:min: ${start_time_hr}:${start_time_minutes} ${start_time_ampm} end time hr:min ${end_time_hr}:${end_time_minutes} ${end_time_ampm}`);
+            
+            // Constant attributes
+            event_label.style.position = 'absolute';
+            event_label.style.left = 0;
+            event_label.style.color = 'white';
+            event_label.style.display = 'flex';
+            event_label.style.flexDirection = 'row';
+            // event_label.style.width = '100%';
+            event_label.style.justifyContent = 'center';
+            event_label.style.alignItems = 'center';
+            event_label.style.backgroundColor = 'lightpink';
+            event_label.textContent = allEve[j].content;
+            event_label.style.border = '1px solid black';
+            
+            // position: absolute;
+            // display: flex;
+            // flex-direction: row;
+            // justify-content: center;
+            // align-items: center;
+            // width: 33%;
+            /*
+                Use start time to calculate 'top'
+            */
+            let top = start_time_hr + 10*(start_time_minutes/60);
+            if(start_time_ampm == 'AM') { 
+                top += '%';
+                event_label.style.top = top; 
+            } else {
+                top = parseFloat(top) + 120;
+                top += '%';
+                event_label.style.top = top;
+            }
+
+            /* 
+                Use end time to calculate 'bottom'
+                Note: this depends on whether event ends before or after 10 am
+                less than  10 and AM:
+            */
+            let bottom;
+            if(end_time_hr < 10 && end_time_ampm == 'AM') { 
+                bottom = 10*(10-end_time_hr) + 10*(end_time_minutes/60);
+                bottom += '%';
+                event_label.style.bottom = bottom; 
+            } else if(end_time_hr > 10 && end_time_ampm == 'PM') {
+                bottom = 10 + 10*(end_time_minutes/60);
+                bottom = '-' + bottom + '%';
```

Comment:
Codacy has a fix for the issue: [Unexpected string concatenation.](https://app.codacy.com/gh/cse110-sp21-group31/cse110-sp21-group31/pullRequest?prid=7575454)
```suggestion
                bottom = `-${  bottom  }%`;
```